docs/libs/rest-routes-example.md

```diff
@@ -0,0 +1,153 @@
+---
+id: libs-features-rest-route-example
+title: Route Example
+---
+
+To implement the new Rest API Route, you would need to do a few things, but for this example, we will use this folder structure:
+* src
+  * class-main.php
+  * rest
+    * class-example-route.php
+
+## Follow these steps:
+* Create a new folder to put this class. For example, `Rest`. Or put this in a place you think it would suit your project best.
+* Create a new class with the same name as your class. For this example, this would be `class-example-route.php`.
+* Copy this example and update it to your needs.
+* Register this new class in `clas-main.php` inside `get_service_classes` method.
+* In terminal run `composer dump-autoload`.
+
+
+### Class-example-route.php
+
+```php
+<?php
+/**
+ * The class register route for example endpoint
+ *
+ * @since   1.0.0
+ * @package Eightshift_Boilerplate\Rest
+ */
+
+namespace Eightshift_Boilerplate\Rest;
+
+use Eightshift_Libs\Rest\Base_Route;
+use Eightshift_Libs\Rest\Callable_Route;
+use Eightshift_Libs\Core\Config_Data;
+
+/**
+ * Class Example_Route
+ */
+class Example_Route extends Base_Route implements Callable_Route {
+
+  /**
+   * Route slug
+   *
+   * @since 1.0.0
+   */
+  const ENDPOINT_SLUG = '/custom-route-slug';
+
+  /**
+   * Instance variable of project config data.
+   *
+   * @var object
+   *
+   * @since 1.0.0
+   */
+  protected $config;
+
+  /**
+   * Create a new instance that injects classes
+   *
+   * @param Config_Data $config Inject config which holds data regarding project details.
+   *
+   * @since 1.0.0 Init.
+   */
+  public function __construct( Config_Data $config ) {
+    $this->config = $config;
+  }
+
+  /**
+   * Method that returns project Route namespace.
+   *
+   * @return string Project namespace for REST route.
+   *
+   * @since 1.0.0 Added in the project
```

Comment:
Since you mentioned `@since` tags, they are actually redundant since we're using git (you can look at at a history of a file easily) and I'd rather if we stopped adding that as it's IMO just noise (Uncle Bob Martin approved this msg).